From 4dcb1e62b4b1d0bc36ba811f8b542d44237f1e7e Mon Sep 17 00:00:00 2001
From: chenyanxzhu <chenyanx.zhu@intel.com>
Date: Tue, 15 Nov 2022 15:31:14 +0800
Subject: [PATCH] Add module parameter for virtio2d/3d distinction

The old debugfs way is not working, as debugfs_graphics related
sepolicy rules is deleted. Google does not allow vendor to use
debugfs_graphics in user build from android 12 CTS test.

Tracked-On: OAM-103582
Signed-off-by: chenyanxzhu <chenyanx.zhu@intel.com>
---
 drivers/gpu/drm/virtio/virtgpu_kms.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/virtio/virtgpu_kms.c b/drivers/gpu/drm/virtio/virtgpu_kms.c
index 21f410901694..d0248413d9db 100644
--- a/drivers/gpu/drm/virtio/virtgpu_kms.c
+++ b/drivers/gpu/drm/virtio/virtgpu_kms.c
@@ -31,6 +31,11 @@
 
 #include "virtgpu_drv.h"
 
+static bool virtio_gpu_is_virgl_3d = false;
+
+MODULE_PARM_DESC(is_virgl_3d, "Virgl 3d switch");
+module_param_named(is_virgl_3d, virtio_gpu_is_virgl_3d, bool, 0444);
+
 static void virtio_gpu_config_changed_work_func(struct work_struct *work)
 {
 	struct virtio_gpu_device *vgdev =
@@ -157,8 +162,12 @@ int virtio_gpu_init(struct drm_device *dev)
 	spin_lock_init(&vgdev->obj_free_lock);
 
 #ifdef __LITTLE_ENDIAN
-	if (virtio_has_feature(vgdev->vdev, VIRTIO_GPU_F_VIRGL))
+	if (virtio_has_feature(vgdev->vdev, VIRTIO_GPU_F_VIRGL)) {
 		vgdev->has_virgl_3d = true;
+		virtio_gpu_is_virgl_3d = true;
+	} else {
+		virtio_gpu_is_virgl_3d = false;
+	}
 #endif
 	if (virtio_has_feature(vgdev->vdev, VIRTIO_GPU_F_EDID)) {
 		vgdev->has_edid = true;
-- 
2.39.2

